src/Common/FitOnFhir.Common/JwtSecurityTokenHandlerProvider.cs

```diff
@@ -0,0 +1,38 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.IdentityModel.Tokens.Jwt;
+using Microsoft.IdentityModel.Tokens;
+
+namespace FitOnFhir.Common
+{
+    public class JwtSecurityTokenHandlerProvider : IJwtSecurityTokenHandlerProvider
+    {
+        private JwtSecurityTokenHandler _jwtSecurityTokenHandler;
+
+        public JwtSecurityTokenHandlerProvider()
+        {
+            _jwtSecurityTokenHandler = new JwtSecurityTokenHandler();
+        }
+
+        /// <inheritdoc/>
+        public JwtSecurityToken ReadJwtToken(string token)
+        {
+            return _jwtSecurityTokenHandler.ReadJwtToken(token);
```

Comment:
Do we want to do `EnsureArg.IsNotNullOrWhiteSpace(token, nameof(token));` and also add null checks for the method below?